### ISSUE TITLE
Feature Request: Generate and display quest flowchart

### ISSUE THREAD

**Issue by PISTOLCUPCAKES** [TarkovTracker/TarkovTracker#382]
It would be really nice if the tracker could dynamically generate and display a quest flowchart similar to [this out of date one on the wiki](https://static.wikia.nocookie.net/escapefromtarkov_gamepedia/images/1/1f/Tarkov_Quest_Flow_Chart_TheDesertFox19.png/revision/latest?cb=20230301195408)

It should either hide completed quests or somehow mark/indicate that they are completed. This would greatly help in visualizing quest progress and identifying bottlenecks (i.e., a quest blocking downstream quests/progress), as well as prioritizing which quests to focus on.

**Comment by thaddeus** [comment]
Hey! Thanks for opening an issue for this. This is something I've considered in the past, but there are two design issues I haven't figured out how I should solve.

The first being that the tree would be big enough that you would need to scroll to effectively show the entire tree. So far I haven't seen an open source tree view component that would make implementing the custom display elements I want from this well, and be responsive. There probably exists some good ones that would work well, I'm just not aware of them. If anyone out there has good suggestions, feel free to leave them here!

The second issue is less technical, but isn't simple. The current task trees are currently fairly interweaved. There are multiple roots from different traders, and later down the tree some tasks rely on completions from multiple other traders. The result is a fairly large tree where fitting the relevant information on the screen could probably be done 80% of the time reliably. The other 20% of the time you would have to navigate around the tree to see the relevant information. While this would still be a cool way to explore the tasks and their relations, I would prefer to have a good design in mind on how to display the relevant bits all in one view. Perhaps that just means implementing the trees on task detail pages showing only the parent and children nodes, and ignoring all the cousin nodes. I'm also open to suggestions on this!

**Comment by PISTOLCUPCAKES** [comment]
Yeah I completely understand that it may be messy and large, but in my opinion that is just all the more reason to implement something like this. I find it infinitely easier to look at tarkov's quests in a flowchart format to easily see which ones I care about completing. I did try to find some libraries a few months ago when I was interested in this and I think [GoJS](https://gojs.net/latest/samples/flowchart.html) seemed like a pretty solid solution, although I only got as far as a simple proof of concept.

As for your second issue, perhaps it could be broken down into one flowchart per trader and where there are prerequisite quests from other traders, you only show the first upstream quest (maybe in a different color, or greyed out, or some other indication that its from another tree). That should help simplify the tree, at least a little.

**Comment by DysektAI** [comment]
## Has your issue been resolved? @PISTOLCUPCAKES 
- **Yes** → please close this issue (use **Close issue** or **Close with comment** below).
- **No** → leave a short comment so others know it's still a problem.

> [!IMPORTANT]
> This repository and [TarkovTracker.io](https://TarkovTracker.io) are **currently unmaintained**.  
> It’s unlikely that new issues or pull requests will receive a response from the original maintainers.

### 👉 Looking for the active project?
- **Website:** [TarkovTracker.org](https://TarkovTracker.org)
- **Source code:** [tarkovtracker-org/TarkovTracker](https://github.com/tarkovtracker-org/TarkovTracker)
- **Open a new issue here:** https://github.com/tarkovtracker-org/TarkovTracker/issues/new/choose
- **Project status & changelog:** #455 

_Thanks for keeping things tidy and for using TarkovTracker!_
